client/web/web.go

```diff
@@ -65,13 +65,32 @@ type Server struct {
 	browserSessions sync.Map
 }
 
+// ServerType specifies the type of a web.Server.
+// If "tailscaled", a full management client is being served.
+// If "system-app", a readonly login client is being served.
+type ServerType string
```

Comment:
ServerType seems an odd name for this, especially with these values.  I guess they're more about the environment in which the server is running (tailscaled vs system-app, which is also a little confusing)  We've previously been referring to this as a "client mode" or similar. 🤔

Reply:
Yeah I was going back and forth between:
* "ClientMode" w/ values "login"/"management"/"legacy" (or similar)
* "ServerType" w/ values "system-app"/"tailscaled"/"legacy"

I did like having "Server" in the name because it specifies the type of app we want to serve. Beyond that, I don't have crazy conviction for the values. I think we agree there's three modes, but we need a good way to name them.

Reply:
I think the values I currently specified are more like "ServerHost" values

Reply:
what about a hybrid of those... `ServerMode` with values "login / manage / legacy" ?  That way we're explicitly stating what mode we want the server to be running in, with values that are consistent with how we've been describing things to date.

Reply:
okay, lets go with that for now - if we come up with something better we can change it later